enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,430 @@
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+	// Defines the metrics collection profile that Prometheus uses to collect
+	// metrics from the platform components. Supported values are `full` or
+	// `minimal`. In the `full` profile (default), Prometheus collects all
+	// metrics that are exposed by the platform components. In the `minimal`
+	// profile, Prometheus only collects metrics necessary for the default
+	// platform alerts, recording rules, telemetry and console dashboards.
+	CollectionProfile CollectionProfile `json:"collectionProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric can be excluded from the `minimal` profile or
+not. To aid teams with this effort the monitoring team will provide:
+- a CLI tool that consumes a ServiceMonitor and a list of metrics and
+  generates the `minimal` ServiceMonitor for that operator.
+- an origin/CI test that validates for all Alerts and PrometheusRules that the
+  metrics used by them are present in the `keep` expression of the
+  ServiceMonitor for the `minimal` profile
+
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
+  that metric gets updated?
+  - The origin/CI test mentioned in the previous section will fail if there is a
+    resource (Alerts/PrometheusRules/Dashboards) using a metric which is not
+    present in the ServiceMonitor;
+
+- A new profile is added, what will happen to operators that had implemented
+  metrics collection profiles but did not implement the latest profile.
+  - The monitoring team will use the list under 
+  [Infrastruture needed](#infrastructure-needed-optional) to help the developers
+  with the addoption of the new metrics collection profile.
+  - Furthermore, we plan on adding extra origin/CI tests to validate each new
+    profile if applicable.
+
+- What happens if a user provides an invalid value for a matrics collection profile?
+  - CMO will reconcile and validate that the value supplied is invalid and it
+    will report Degraded=False and fail reconciliation.
+
+- Should we add future profiles? How would we validate such profiles?
+  - Our current validation strategy with only two profiles is quite linear,
+    however, things start becoming more complex and hard to mainain as we
+    introduce new profiles to the mix. 
+  - Some of the things to consider if new profiles are introduce are:
+      - How would we validate such profile?
+      - How would we ensure teams that adopted metrics collection profiles implement the new profile?
```

Comment:
[`cpv#impl-stats.go`](https://github.com/rexagod/cpv/blob/main/internal/profiles/impl-stats.go) can report partially implemented collection profiles.

Reply:
Feel free to resolve or add the aforementioned link.